lib/compile.js

```diff
@@ -175,6 +198,11 @@ function compiler() {
 		throw new Error( "No formatters or parsers has been provided" );
 	}
 
+	if ( args.length && typeof args[args.length - 1] === "object" &&
+			typeof args[args.length - 1].template === "function" ) {
+		templateFn = args[args.length - 1].template;
+	}
+
```

Comment:
I think we should simplify the API. Let's abort the `compiler( formatterOrParser, ... )` and stick with what you have documented.